### PR TITLE
[REF] base: ir_ui_view get_view_arch_from_file

### PR DESCRIPTION
**Pull request by Julien00859**
I had troubles understanding how was the function working, I decided to
rework it a bit, ended up rewritting much of it. I hope this new
implementation is easier to understand.

Task: 2463632

### CODE REVIEW

**Review by rco-odoo — APPROVED** [review]
@robodoo delegate+

File: odoo/addons/base/models/ir_ui_view.py [hunk 1]
```diff
@@ -154,42 +154,49 @@ def _hasclass(context, *cls):
...
+    # for views without this prefix.
+    if view_id.endswith('_ir_ui_view'):
+        # len('_ir_ui_view') == 11
+        xpath = xpath[:-1] + f" or @id='{xmlid[:-11]}' or @id='{view_id[:-11]}']"
+
+    document = etree.parse(filepath)
+    for node in document.xpath(xpath):
+        if node.tag == 'record':
+            field_arch = node.find('field[@name="arch"]')
+            if field_arch:
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_ui_view.py:172]:
```suggestion
            if field_arch is not None:
```

File: odoo/addons/base/models/ir_ui_view.py [hunk 2]
```diff
@@ -154,42 +154,49 @@ def _hasclass(context, *cls):
...
+                _fix_multiple_roots(field_arch)
+                inner = ''.join(
+                    etree.tostring(child, encoding='unicode')
+                    for child in field_arch.iterchildren()
+                )
+                return field_arch.text + inner
+
+            field_view = node.find('field[@name="view_id"]')
+            if field_view:
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_ui_view.py:181]:
```suggestion
            if field_view is not None:
```

**Comment by Julien00859** [comment]
@robodoo r+

**Pull request closed without merging**